src/ln/channelmonitor.rs

```diff
@@ -82,6 +84,17 @@ pub enum ChannelMonitorUpdateErr {
 #[derive(Debug)]
 pub struct MonitorUpdateError(pub &'static str);
 
+/// Simple structure send back by ManyChannelMonitor in case of HTLC detected onchain from a 
+/// forward channel and from which info are needed to update HTLC in a backward channel.
+/// If payment_preimage is None, htlc output has been timeout.
+/// If HTLCSource is None, htlc was an inbound one, we don't have to pass state backward
+/// but we may interested by the fact htlc output has been solved
+pub struct HTLCUpdate {
+	pub(super) payment_hash: PaymentHash,
+	pub(super) payment_preimage: Option<PaymentPreimage>,
+	pub(super) source: Option<HTLCSource>
```

Comment:
Yes, I know that None HTLCSource implies inbound HTLC but I voluntary let them because user may be interested in logging that a given inbound HTLC  has been resolved even if right now we do nothing with them after fetch_pending_htlc_updated ?

Reply:
Okay remove Option<> and just log_trace them if we detect them in is_resolving_htlc_output